docs/setup/independent/high-availability.md

```diff
@@ -0,0 +1,328 @@
+```
+
+This should result in `client.pem` and `client-key.pem` being created.
+
+### set environment vars
+
+1. Open new new tabs in your shell for `etcd1` and `etcd2`. Ensure you are SSHed into all three VMs are run the following (it will be a lot quicker if you use tmux syncing - to do this in iTerm enter `cmd+shift+i`):
+
+```
+export PEER_NAME=$(hostname)
+export PUBLIC_IP=$(ip addr show eth0 | grep -Po 'inet \K[\d.]+')
+export PRIVATE_IP=$(ip addr show eth1 | grep -Po 'inet \K[\d.]+')
+
+touch /etcd/etcd.env
+echo "PEER_NAME=$PEER_NAME" >> /etcd/etcd.env
+echo "PUBLIC_IP=$PUBLIC_IP" >> /etcd/etcd.env
+echo "PRIVATE_IP=$PRIVATE_IP" >> /etcd/etcd.env
+```
+
+### create ssh access
+
+1. Your `etcd1` and `etcd2` VMs need access to the CA certs created by `etcd0`. To do this, we'll use SCP which relies on SSH. So let's generate some SSH keys for the boxes:
+
+```
+ssh-keygen -t rsa -b 4096 -C ""
+```
+
+Keep hitting enter until files exist in `~/.ssh`.
+
+1. Find the public key for `etcd1` and `etcd2` like so:
+
+```
+cat ~/.ssh/id_rsa.pub
+```
+
+1. Copy the output for each VM and paste into `etcd0`'s `~/.ssh/authorized_keys` file. This will permit `etcd1` and `etcd2` to SSH in to the VM.
+
+### generate the certs
+
+1. On `etcd1` and `etcd2`, run the following:
+
+```
+mkdir ~/certs
+cd ~/certs
+scp root@<etcd1-ip-address>:/root/certs/ca.pem .
+scp root@<etcd1-ip-address>:/root/certs/ca-key.pem .
+```
+
+1. On all VMs, run the following:
+
+```
+cfssl print-defaults csr > config.json
+sed -i '0,/CN/{s/example\.net/'"$PEER_NAME"'/}' config.json
+sed -i 's/www\.example\.net/'"$PRIV_IP"'/' config.json
+sed -i 's/example\.net/'"$PUB_IP"'/' config.json
+
+cfssl gencert -ca=ca.pem -ca-key=ca-key.pem -config=ca-config.json -profile=server config.json | cfssljson -bare server
+cfssl gencert -ca=ca.pem -ca-key=ca-key.pem -config=ca-config.json -profile=peer config.json | cfssljson -bare peer
+```
+
+This will result in the following files at a minimum: `peer.pem`, `peer-key.pem`, `server.pem`, `server-key.pem`.
+
+### set up systemd
+
+1. Now that all the certificates have been generated, you will now install and set up etcd on each VM. First install etcd binaries like so:
+
+```
+export ETCD_VERSION=v3.2.8
+wget -q --show-progress --https-only --timestamping \
+  "https://github.com/coreos/etcd/releases/download/$ETCD_VERSION/etcd-$ETCD_VERSION-linux-amd64.tar.gz"
+tar -xvf etcd-$ETCD_VERSION-linux-amd64.tar.gz
+sudo mv etcd-$ETCD_VERSION-linux-amd64/etcd* /usr/local/bin/
+rm -rf etcd-v3.2.10-linux-amd64*
+```
+
+1. Next, copy the systemd unit file like so:
+
+```
+cat >/etc/systemd/system/etcd.service <<EOL
+[Unit]
+Description=etcd
+Documentation=https://github.com/coreos/etcd
+Conflicts=etcd.service
+Conflicts=etcd2.service
+
+[Service]
+EnvironmentFile=/etc/etcd.env
+Type=notify
+Restart=always
+RestartSec=5s
+LimitNOFILE=40000
+TimeoutStartSec=0
+
+ExecStart=/usr/local/bin/etcd --name ${PEER_NAME} \
```

Comment:
why not run this as a Static Pod? That'd be way better IMO

Reply:
because this is on a node without k8s on, it's dedicated to just etcd. I think the whole point of an off-cluster etcd is being decoupled from the system it's persisting

Reply:
also worth noting that systemd etcd is (AFAIK) how coreos recommend production etcd to be run